proxy/src/control/destination/mod.rs

```diff
@@ -113,6 +161,7 @@ impl Resolver {
 
         Resolution {
             update_rx,
+            _active: weak_active,
```

Comment:
This seems backwards? I expected this to work like so:

We want to know when an object has been dropped. That object should have an `Arc<()>` that it doesn't do anything with except drop when it itself is dropped.

Then the other side will hold a `Weak<()>`. Every time it tries to use the `Weak<()>` it has to upgrade it to an `Arc<>`. That upgrade will succeed if the object(s) on the other side haven't all been dropped yet. But, as soon as the other side is dropped, there will be no more `Arc<>`s and then upgrading the `Weak<()>` to an `Arc<()>` will return `None`. When that happens we know we can drop the destination.

Reply:
ah, yes much nicer.